bpo-36346: Add Py_DEPRECATED to deprecated unicode APIs (#20878)
python/cpython: Misc/NEWS.d/next/C API/2020-06-17-11-24-00.bpo-36346.fTMr3S.rst

Mark ``Py_UNICODE_COPY``, ``Py_UNICODE_FILL``, ``PyUnicode_WSTR_LENGTH``,
``PyUnicode_FromUnicode``, ``PyUnicode_AsUnicode``, ``_PyUnicode_AsUnicode``,

> and ``PyUnicode_AsUnicodeAndSize`` deprecated in C. Remove ``Py_UNICODE_MATCH``
> because it was broken since Python 3.3.

Reviewer: ```suggestion
and ``PyUnicode_AsUnicodeAndSize`` as deprecated in C. Remove ``Py_UNICODE_MATCH``
because it has been broken since Python 3.3.
```

> and ``PyUnicode_AsUnicodeAndSize`` as deprecated in C. Remove ``Py_UNICODE_MATCH``
> because it has been broken since Python 3.3.

Reviewer 2: ```suggestion
and ``PyUnicode_AsUnicodeAndSize`` as deprecated in C. Remove ``Py_UNICODE_MATCH``
which was deprecated and broken since Python 3.3.
```
